onnxruntime/core/providers/cuda/cuda_provider_factory.cc

```diff
@@ -279,10 +279,108 @@ struct CUDA_Provider : Provider {
     DeleteRegistry();
   }
 
+  Status CreateIExecutionProvider(const OrtHardwareDevice* const* /*devices*/,
+                                  const OrtKeyValuePairs* const* /*ep_metadata*/,
+                                  size_t num_devices,
+                                  ProviderOptions& provider_options,
+                                  const OrtSessionOptions& session_options,
+                                  const OrtLogger& logger,
+                                  std::unique_ptr<IExecutionProvider>& ep) override {
+    OrtCUDAProviderOptionsV2 options;
+    UpdateProviderOptions(&options, provider_options);
+    auto ep_factory = CreateExecutionProviderFactory(&options);
+    ep = ep_factory->CreateProvider(session_options, logger);
+
+    return Status::OK();
+  }
+
 } g_provider;
 
 CUDA_Provider* GetProvider() {
   return &g_provider;
 }
 
 }  // namespace onnxruntime
+
+#include "core/framework/error_code_helper.h"
+
+// OrtEpApi infrastructure for
+struct CudaEpFactory : OrtEpFactory {
+  CudaEpFactory(const OrtApi& ort_api_in) : ort_api{ort_api_in} {
+    GetName = GetNameImpl;
+    GetVendor = GetVendorImpl;
+    GetDeviceInfoIfSupported = GetDeviceInfoIfSupportedImpl;
+    CreateEp = CreateEpImpl;
+    ReleaseEp = ReleaseEpImpl;
+  }
+
+  static const char* GetNameImpl(const OrtEpFactory* this_ptr) {
+    const auto* factory = static_cast<const CudaEpFactory*>(this_ptr);
+    return factory->ep_name.c_str();
+  }
+
+  static const char* GetVendorImpl(const OrtEpFactory* this_ptr) {
+    const auto* factory = static_cast<const CudaEpFactory*>(this_ptr);
+    return factory->vendor.c_str();
+  }
+
+  static bool GetDeviceInfoIfSupportedImpl(const OrtEpFactory* this_ptr,
+                                           const OrtHardwareDevice* device,
+                                           _Out_opt_ OrtKeyValuePairs** ep_metadata,
+                                           _Out_opt_ OrtKeyValuePairs** ep_options) {
+    const auto* factory = static_cast<const CudaEpFactory*>(this_ptr);
+
+    if (factory->ort_api.HardwareDevice_Type(device) == OrtHardwareDeviceType::OrtHardwareDeviceType_GPU &&
+        factory->ort_api.HardwareDevice_VendorId(device) == 0x10de) {
+      return true;
+    }
+
+    return false;
```

Comment:
Why don't we just return the expression result?

Reply:
I expect the 'real' implementation should check if the device has the required compute capability. this is essentially a stub and any additional selection logic needs to be added in the `if` branch, which may include populating metadata or provider option info.